packages/contracts/src/dollar/core/StakingShare.sol

```diff
@@ -134,15 +152,72 @@ contract StakingShare is ERC1155Ubiquity {
         return _stakes[id];
     }
 
+    /**
+     * @dev See {IERC1155-safeBatchTransferFrom}.
+     */
+    function safeBatchTransferFrom(
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    ) public virtual override(ERC1155, ERC1155Ubiquity) whenNotPaused {
+        super.safeBatchTransferFrom(from, to, ids, amounts, data);
+        _holderBalances[to].add(ids);
+    }
+
+    function _burnBatch(
```

Comment:
We should remove this method because it is already implemented here https://github.com/ubiquity/ubiquity-dollar/blob/4b1e1516df5bc41b68c0fcf9083743122fb32933/packages/contracts/src/dollar/core/ERC1155Ubiquity.sol#L185